monero/client_test.go

```diff
@@ -84,20 +79,20 @@ func TestClient_Transfer(t *testing.T) {
 		time.Sleep(time.Second)
 	}
 
-	_ = daemon.callGenerateBlocks(xmrmakerAddr.Address, 16)
+	err = daemon.callGenerateBlocks(xmrmakerAddr.Address, 16)
+	require.NoError(t, err)
 
 	// generate spend account for A+B
 	skAKPriv := mcrypto.SumPrivateSpendKeys(kpA.SpendKey(), kpB.SpendKey())
 	// ignore the error for now, as it can error with "Wallet already exists."
-	_ = cXMRTaker.callGenerateFromKeys(skAKPriv, vkABPriv, kpABPub.Address(common.Mainnet),
-		fmt.Sprintf("test-wallet-%d", r), "")
+	_ = cXMRTaker.callGenerateFromKeys(skAKPriv, vkABPriv, kpABPub.Address(common.Mainnet), walletFP, "")
 
 	err = cXMRTaker.refresh()
 	require.NoError(t, err)
 
 	balance, err = cXMRTaker.GetBalance(0)
 	require.NoError(t, err)
-	require.NotEqual(t, 0, balance.Balance)
```

Comment:
I was surprised that this didn't give a mismatched types error.  I guess the test above always passes regardless of the value.  I created this example to understand it better and it passes:
```
func Test_FloatNotZero(t *testing.T) {
	var f = 0.0
	require.NotEqual(t, 0, f)
}
```

Reply:
interesting that it doesn't have a type error actually, good to know!